flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/Dispatcher.java

```diff
@@ -878,8 +879,12 @@ public CompletableFuture<MultipleJobsDetails> requestMultipleJobDetails(Duration
 
                     completedJobDetails.forEach(job -> deduplicatedJobs.put(job.getJobId(), job));
                     runningJobDetails.forEach(job -> deduplicatedJobs.put(job.getJobId(), job));
+                    Collection<JobDetails> orderedDeduplicatedJobs =
+                            deduplicatedJobs.values().stream()
+                                    .sorted(Comparator.comparingLong(JobDetails::getStartTime))
+                                    .collect(Collectors.toList());
 
-                    return new MultipleJobsDetails(new HashSet<>(deduplicatedJobs.values()));
+                    return new MultipleJobsDetails(new ArrayList<>(orderedDeduplicatedJobs));
```

Comment:
```suggestion
                    return new MultipleJobsDetails(orderedDeduplicatedJobs);
```
I guess, we don't need the explicit `ArrayList`, do we?